From 10d41eb3b0a39949d34b9396f2aa0baa2af9e300 Mon Sep 17 00:00:00 2001
From: Bruno Haible <bruno@clisp.org>
Date: Sat, 12 Feb 2022 15:10:41 +0100
Subject: [PATCH 2/3] loongarch: Allow for continuous integration.

* cross-tools/cross-build.sh (func_build_gcc): Download fork tarball
from alpha.gnu.org. Adjust installation completeness test.
* cross-tools/cross.conf (loongarch64): Use binutils 2.38.
---
 ChangeLog                  |  7 +++++++
 cross-tools/cross-build.sh | 15 ++++++++++++---
 cross-tools/cross.conf     |  2 +-
 3 files changed, 20 insertions(+), 4 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index fbd41c8..1b51050 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,10 @@
+2022-02-12  Bruno Haible  <bruno@clisp.org>
+
+	loongarch: Allow for continuous integration.
+	* cross-tools/cross-build.sh (func_build_gcc): Download fork tarball
+	from alpha.gnu.org. Adjust installation completeness test.
+	* cross-tools/cross.conf (loongarch64): Use binutils 2.38.
+
 2022-01-16  Bruno Haible  <bruno@clisp.org>
 
 	loongarch: Add support for loongarch64 ABI.
diff --git a/cross-tools/cross-build.sh b/cross-tools/cross-build.sh
index b015f34..cbca469 100755
--- a/cross-tools/cross-build.sh
+++ b/cross-tools/cross-build.sh
@@ -562,7 +562,16 @@ func_build_gcc ()
     [3-5]*)   pkg_suffix=bz2 ;;
     *)        pkg_suffix=xz ;;
   esac
-  func_ensure_unpacked_source gcc "$version" "$pkg_suffix" "https://ftp.gnu.org/pub/gnu/gcc/gcc-$version" || func_exit 1
+  case "$version" in
+    *-*)
+      # A fork's snapshot.
+      func_ensure_unpacked_source gcc "$version" "$pkg_suffix" "https://alpha.gnu.org/gnu/libffcall/gcc-$version" || func_exit 1
+      ;;
+    *)
+      # An official GNU release
+      func_ensure_unpacked_source gcc "$version" "$pkg_suffix" "https://ftp.gnu.org/pub/gnu/gcc/gcc-$version" || func_exit 1
+      ;;
+  esac
   if test "$target" = armv7l-linux-gnueabihf; then
     configure_options="$configure_options --with-arch=armv7-a --with-float=hard --with-fpu=vfpv3-d16"
   fi
@@ -588,8 +597,8 @@ func_build_gcc ()
   # gcc < 3.4 uses lib/gcc-lib/${gcctarget}/${version}/ whereas
   # gcc >= 3.4 uses libexec/gcc/${gcctarget}/${version}/.
   if { test -f "$HOST_CROSS_DIR/${target}-tools/bin/${target}-gcc" \
-       && { test -f "$HOST_CROSS_DIR/${target}-tools/libexec/gcc/${gcctarget}/${version}/cc1" \
-            || test -f "$HOST_CROSS_DIR/${target}-tools/lib/gcc-lib/${gcctarget}/${version}/cc1"; }; \
+       && { test -f "$HOST_CROSS_DIR/${target}-tools/libexec/gcc/${gcctarget}/`echo ${version} | sed -e 's/-.*//'`/cc1" \
+            || test -f "$HOST_CROSS_DIR/${target}-tools/lib/gcc-lib/${gcctarget}/`echo ${version} | sed -e 's/-.*//'`/cc1"; }; \
      }; then
     echo "Installation of gcc is complete."
   else
diff --git a/cross-tools/cross.conf b/cross-tools/cross.conf
index a684540..b6eccde 100644
--- a/cross-tools/cross.conf
+++ b/cross-tools/cross.conf
@@ -24,4 +24,4 @@ s390         s390-linux              3.1          2.12.90.0.7  s390-linux
 s390x        s390x-linux             5.4.0        2.24         s390x-linux
 riscv32      riscv32-linux           7.3.0        2.30         riscv32-linux
 riscv64      riscv64-linux           7.3.0        2.30         riscv64-linux
-loongarch64  loongarch64-linux       12.0.0-loong 2.37-loong   loongarch64-linux
+loongarch64  loongarch64-linux       12.0.0-loong 2.38         loongarch64-linux
-- 
2.20.1

